src/Dfc.CourseDirectory.WebV2/Filters/ProviderContextResourceFilter.cs

```diff
@@ -100,6 +112,17 @@ async Task TryAssignFeature()
 
                 return null;
             }
+
+            async Task<Guid?> TryGetProviderIdFromLegacyContext()
+            {
+                var ukprn = context.HttpContext.Session.GetInt32("UKPRN");
+                if (ukprn.HasValue)
```

Comment:
Consider return `ukprn.HasProvider ? await providerInfoCache.GetProviderIdForUkprn(ukprn.Value) : null` for succinctness